### PR TITLE
Quite possibly the smallest microoptimization here

### PR DESCRIPTION
**Pull request by Cyberboss**

### CODE REVIEW

File: code/controllers/subsystem.dm [hunk 1]
```diff
@@ -145,9 +145,10 @@
 
 /datum/controller/subsystem/proc/pause()
 	. = 1
-	if (state == SS_RUNNING)
+	var/current_state = state
+	if (current_state == SS_RUNNING)
 		state = SS_PAUSED
-	else if (state == SS_SLEEPING)
+	else if (current_state == SS_SLEEPING)
```
Comment (coiax) [inline code/controllers/subsystem.dm]: wouldn't a switch statement be better? idk about microoptimisations

**Comment by xxalpha** [comment]
How is this an optimization at all? I don't get it.

**Comment by Cyberboss** [comment]
Accessing proc vars is faster than datum vars

**Comment by xxalpha** [comment]
Oh it's that stuff. Did you profile it?

**Comment by Cyberboss** [comment]
No need, it's simply reducing 2 reads to one

**Comment by YotaXP** [comment]
Are two member variable reads really slower than one member variable read + one local variable assignment + two local variable reads?

**Comment by Cyberboss** [comment]
Yes, because datums have O(logn) lookup time where N is the number of vars
they have. The stack has constant time

File: code/controllers/subsystem.dm (hunk 1 above)
Comment (lzimann) [inline code/controllers/subsystem.dm]: Yea do a switch here.

**Comment by duncathan** [comment]
datum var lookups are stupid slow

**Pull request merged**